varcode/variant.py

```diff
@@ -175,71 +197,86 @@ def coding_genes(self):
+        # Intergenic effect
+        # TODO: look for nearby genes and mark those as Upstream and Downstream
+        # effects
+        if len(gene_ids) == 0:
+            return [Intergenic(self)]
 
+        overlapping_transcripts = self.transcripts()
         # group transcripts by their gene ID
-        overlapping_transcript_groups = group_by(
-            overlapping_transcripts, field_name='gene_id')
-
-        # dictionary from gene ID to list of transcript effects
-        gene_transcript_effects_groups = {}
-
-        # mapping from Transcript objects to errors encountered
-        # while trying to annotated them
-        errors = {}
-
-        for (gene_id, transcripts) in overlapping_transcript_groups.items():
-            effects = []
-            for transcript in transcripts:
-                try:
-                    effects.append(self.transcript_effect(transcript))
-                except (AssertionError, ValueError) as error:
-                    if raise_on_error:
-                        raise
-                    else:
-                        logging.warn(
-                            "Encountered error annotating %s for %s: %s",
-                            self,
-                            transcript,
-                            error)
-                    errors[transcript] = error
-            gene_transcript_effects_groups[gene_id] = effects
-
-        return VariantEffectCollection(
-            variant=self,
-            gene_effect_groups=gene_transcript_effects_groups,
-            errors=errors)
-
-    def transcript_effect(self, transcript):
+        transcripts_grouped_by_gene = groupby_field(
+            overlapping_transcripts, 'gene_id')
+
+        # list of all MutationEffects for all genes & transcripts
+        effects = []
+
+        # want effects in the list grouped by the gene they come from
+        for gene_id in sorted(gene_ids):
+            if gene_id not in transcripts_grouped_by_gene:
+                # intragenic variant overlaps a gene but not any transcripts
+                gene = self.ensembl.gene_by_id(gene_id)
+                effects.append(Intragenic(self, gene))
+            else:
+                # gene ID  has transcripts overlapped by this variant
+                for transcript in transcripts_grouped_by_gene[gene_id]:
+                    try:
+                        effect = self.effect_on_transcript(transcript)
+                        effects.append(effect)
+                    except (AssertionError, ValueError) as error:
+                        if raise_on_error:
+                            raise
+                        else:
+                            logging.warn(
+                                "Encountered error annotating %s for %s: %s",
+                                self,
+                                transcript,
+                                error)
+        return effects
+
+    @memoize
+    def transcript_effect_dict(self, *args, **kwargs):
```

Comment:
`VariantEffectCollection` is gone on this branch, and all of its methods and properties (like this one) have migrated onto `VariantCollection`.